### PR TITLE
feat(prometheus_exporter sink): Add an option to allow suppressing timestamp

### PR DESCRIPTION
**Pull request by shenxn**
This PR adds an option `expose_timestamp` that allows users to control whether to render metric timestamp in the response. Closes https://github.com/vectordotdev/vector/issues/6724 and related to https://github.com/vectordotdev/vector/issues/7350

### CODE REVIEW

**Review by bruceg — COMMENTED** [review]
Thanks for this contribution, @shenxn. This code looks good. I wonder if we want the negative in the configuration, as in `strip_timestamp` or similar, since that better expresses what the effect of this is. There's also a typo in the added `.cue` documentation as can be [seen here](https://github.com/vectordotdev/vector/runs/7076945617?check_suite_focus=true#step:16:31).

**Comment by jszwedko** [comment]
Yeah, we recently added a `supress_timestamp` option to the `splunk_hec_logs` sink that is similar to this so I think reusing that naming would be good for consistency (and inverting the default).

Thanks for this contribution @shenxn !

File: src/sinks/prometheus/exporter.rs [hunk 1]
```diff
@@ -75,6 +75,8 @@ pub struct PrometheusExporterConfig {
     #[serde(default = "default_flush_period_secs")]
     #[serde_as(as = "serde_with::DurationSeconds<u64>")]
     pub flush_period_secs: Duration,
+    #[serde(default = "default_suppress_timestamp")]
```
Comment (spencergilbert) [inline src/sinks/prometheus/exporter.rs:78]:
```suggestion
    #[serde(default)]
```

IIRC the default here is `false`, so we could drop the function as well?

**Review by jszwedko — APPROVED** [review]
Great, thanks for this contribution @shenxn !

**Review by bruceg — APPROVED** [review]

**Comment by xiaonanshen-ponyai** [comment]
Any ideas on why the soak analysis checks failed? Seems all of them failed at the same step:
```
[open_capture] reading: 2580221269-1-captures/2580221269-1-syslog_log2metric_humio_metrics-1/comparison/0/comparison.captures
Traceback (most recent call last):
  File "/home/runner/actions-runner/_work/vector/vector/./soaks/bin/plot_analysis", line 21, in <module>
    bytes_written = pd.concat(common.compute_throughput(
  File "/opt/hostedtoolcache/Python/3.10.2/x64/lib/python3.10/site-packages/pandas/util/_decorators.py", line 311, in wrapper
    return func(*args, **kwargs)
  File "/opt/hostedtoolcache/Python/3.10.2/x64/lib/python3.10/site-packages/pandas/core/reshape/concat.py", line 347, in concat
    op = _Concatenator(
  File "/opt/hostedtoolcache/Python/3.10.2/x64/lib/python3.10/site-packages/pandas/core/reshape/concat.py", line 401, in __init__
    objs = list(objs)
  File "/home/runner/actions-runner/_work/vector/vector/soaks/bin/common.py", line 43, in compute_throughput
    for capture in captures:
  File "/home/runner/actions-runner/_work/vector/vector/soaks/bin/common.py", line 39, in open_captures
    yield pd.concat(open_capture(f, metric_name, unwanted_labels))
  File "/opt/hostedtoolcache/Python/3.10.2/x64/lib/python3.10/site-packages/pandas/util/_decorators.py", line 311, in wrapper
    return func(*args, **kwargs)
  File "/opt/hostedtoolcache/Python/3.10.2/x64/lib/python3.10/site-packages/pandas/core/reshape/concat.py", line 347, in concat
    op = _Concatenator(
  File "/opt/hostedtoolcache/Python/3.10.2/x64/lib/python3.10/site-packages/pandas/core/reshape/concat.py", line 404, in __init__
    raise ValueError("No objects to concatenate")
ValueError: No objects to concatenate
```

**Comment by jszwedko** [comment]
Thanks @shenxn ! I believe the failing soak test was due needing `master` to be merged in here to bring in updates to its configuration. I did so here.

**Pull request merged**